rule/string-format.go

```diff
@@ -66,12 +67,14 @@ type lintStringFormatRule struct {
 
 type stringFormatSubrule struct {
 	parent       *lintStringFormatRule
-	scope        stringFormatSubruleScope
+	scopes       stringFormatSubruleScopes
 	regexp       *regexp.Regexp
 	negated      bool
 	errorMessage string
 }
 
+type stringFormatSubruleScopes []*stringFormatSubruleScope
```

Comment:
out of curiosity: why is that a slice of pointers? (not a slice of values)

Reply:
I saw one video on youtube where this two approaches were compared and slice of pointers use less memory little bit)